runtime/src/common/crypto/x25519.rs

```diff
@@ -38,6 +38,9 @@ impl PrivateKey {
 impl From<[u8; PRIVATE_KEY_LENGTH]> for PrivateKey {
     /// Load private key from a byte array.
     fn from(bytes: [u8; PRIVATE_KEY_LENGTH]) -> PrivateKey {
+        // We must clamp to match previous x25519-dalek behavior and the test vectors.
+        let bytes = curve25519_dalek::scalar::clamp_integer(bytes);
```

Comment:
Could also change test vectors, as secrets/scalars will always get clamped (when reading in the old version, when doing an operation in the new one).

Reply:
I guess that's true. I was concerned about other use cases, e.g. for the RNG. But that is ephemeral so it may be fine regardless?